x/mongo/driver/topology/connection.go

```diff
@@ -51,6 +51,7 @@ type connection struct {
 	connectContextMade   chan struct{}
 	canStream            bool
 	currentlyStreaming   bool
+	mu                   sync.Mutex
```

Comment:
This could be named a little more descriptively. like connectContextMutex or something?

Reply:
good note. I've changed the name